src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1182 @@
+            logger.info(f"Adding batch {i + 1}")
+            n = graph.submit(
+                fn,
+                work,
+                dataset_uri=dataset_uri,
+                config=config,
+                name=f"{task_prefix} - {i}",
+                mode=tiledb.cloud.dag.Mode.BATCH,
+                append=True,
+                verbose=verbose,
+                trace=trace,
+                log_uri=log_uri,
+                resources=DEFAULT_RESOURCES,
+                image_name=DEFAULT_IMG_NAME,
+                access_credentials_name=acn,
+                **kwargs,
+            )
+
+            n.depends_on(ingest_node)
+            nodes.append(n)
+
+        graph.submit(
+            consolidate_fragment_meta,
+            dataset_uri,
+            config,
+            resources=DEFAULT_RESOURCES,
+            verbose=verbose,
+            log_uri=log_uri,
+            access_credentials_name=acn,
+        ).depends_on(nodes)
+
+        # Register the dataset on TileDB Cloud
+        if register_name:
+            register_dataset_udf(
+                dataset_uri,
+                namespace=namespace,
+                register_name=register_name,
+                config=config,
+                verbose=verbose,
+                trace=trace,
+                log_uri=log_uri,
+                access_credentials_name=acn,
+            )
+
+        run_dag(graph, wait=False)
+
+        logger.info(
+            "%s geospatial datasets ingestion submitted -"
+            " https://cloud.tiledb.com/activity/taskgraphs/%s/%s",
+            dataset_type,
+            graph.namespace,
+            graph.server_graph_uuid,
+        )
+    else:
+        # unit test inner functions
+        # set up function
+        fn(
+            dataset_uri=dataset_uri,
+            append=False,
+            verbose=verbose,
+            trace=trace,
+            log_uri=log_uri,
+            **kwargs,
+        )
+        # work functions
+        for i, work in enumerate(batch(samples, batch_size)):
+            fn(
+                work,
+                dataset_uri=dataset_uri,
+                append=True,
+                verbose=verbose,
+                trace=trace,
+                log_uri=log_uri,
+                **kwargs,
+            )
+
+
+def consolidate_fragment_meta(
+    dataset_uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    id: str = "consolidate",
+    verbose: bool = False,
+    log_array_uri: Optional[str] = None,
+) -> None:
+    """
+    Consolidate arrays in the dataset.
+
+    :param dataset_uri: dataset URI
+    :param config: config dictionary, defaults to None
+    :param id: profiler event id, defaults to "consolidate"
+    :param verbose: verbose logging, defaults to False
+    """
+
+    logger = get_logger_wrapper(verbose)
+
+    with tiledb.scope_ctx(config):
+        with Profiler(array_uri=log_array_uri, id=id):
+            modes = ["commits", "fragment_meta", "array_meta"]
```

Comment:
Based on the name of this function (`consolidate_fragment_meta`), check if these are the consolidate/vacuum modes you want.

Reply:
👍 I have renamed the function to that it matches the consolidate/vacuum modes of all metadata.